webapp/src/components/cardDetail/cardDetailContentsUtility.ts

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+import {Position} from './cardDetailContents'
+
+type DragAndDropRearrangeData = {
+    contentOrder: Array<string|string[]>,
+    srcBlockId: string,
+    srcBlockX: number,
+    srcBlockY: number,
+    dstBlockX: number,
+    dstBlockY: number,
+    dstBlockId: string,
+    moveTo: Position,
+}
+
+export const dragAndDropRearrange =
+    (data: DragAndDropRearrangeData): Array<string|string[]> => {
+        const {contentOrder, srcBlockX, srcBlockY, dstBlockX, dstBlockY, moveTo, srcBlockId, dstBlockId} = data
+        const newContentOrder: Array<string|string[]> = JSON.parse(JSON.stringify(contentOrder))
+
+        const copySrcBlockX = srcBlockX
```

Comment:
This won't be a deep copy and only a reference. Will it work?

Reply:
Yep these are primitives (numbers) so it's okay.